compiler.cpp

```diff
@@ -319,18 +319,16 @@ const Strat* ComputeStrategy(const Policy& node, std::unordered_map<const Policy
                     keys.push_back(x.keys[0]);
                 }
                 strats.push_back(MakeStrat(store, Strat::Type::MULTI, std::move(keys), node.k));
-            }
-            if (node.k > 1 && node.k < node.sub.size()) {
-                strats.push_back(MakeStrat(store, Strat::Type::THRESH, subs, node.k, (double)node.k / subs.size()));
-            }
-            if (node.k == 1 || node.k == node.sub.size()) {
+            } else if ((node.k == 1 && node.sub.size() < 10) || node.k == node.sub.size()) {
                 while (subs.size() > 1) {
                     auto rep = MakeStrat(store, node.k == 1 ? Strat::Type::OR : Strat::Type::AND, Vector(*(subs.rbegin() + 1), subs.back()), 1.0 / subs.size());
                     subs.pop_back();
                     subs.pop_back();
                     subs.push_back(MakeStrat(store, Strat::Type::CACHE, Vector(rep)));
                 }
                 strats.push_back(subs[0]);
+            } else {
```

Comment:
I don't think there is a reason for this sequence of if/then/elses. Just use all applicable strategies. The compiler will use the best one.

Reply:
Right. The first `else if` was an oversight, but the `else` part i'm not sure. I figured that it would put unnecessary burden on the compiler to add the thresh strategy if:
- The policy is just a (smallish) multisig
- The policy can be represented as nested `or()`s or `and()`s
as these would always be more efficient than the raw addition of all sub-policies. But maybe my intuition is wrong for the latter?

Anyways, amended :)

Reply:
You may be right, but the code is more obviously going to find the optimal one if it just tries everything.